### PR TITLE
Add basic debug test and modify sim flow

### PR DESCRIPTION
**Pull request by udinator**
- Modified sim.py to correctly process any sim_opts and compare_opts passed specified in the testlist
- Removed shared utility functions from sim.py (duplicates of functions in lib.py)
- Changed location of vseq.stop() call in testbench to prevent debug or interrupt signal assertions after ecall has been detected

### CODE REVIEW

File: dv/uvm/riscv_dv_extension/testlist.yaml [hunk 1]
```diff
@@ -102,7 +102,27 @@
...
+    +no_fence=1
+    +num_of_sub_program=0
+  rtl_test: core_ibex_base_test
+  sim_opts: >
+    +enable_debug_seq=1
+  compare_opts: >
+    compare_final_value_only=1
+    verbose=1
+
+# TODO(udij) - this test is later in testlist - ignore for now
```
Comment (taoliug) [inline dv/uvm/riscv_dv_extension/testlist.yaml]: If it's ignored, should the iteration be set to 0 for now?

Comment (udinator) [reply dv/uvm/riscv_dv_extension/testlist.yaml]: Sure, good point.

File: dv/uvm/sim.py [hunk 2]
```diff
@@ -222,7 +189,17 @@ def compare(test_list, iss, output_dir, verbose):
         else:
           print("Unsupported ISS" % iss)
           sys.exit(1)
-        compare_trace_csv(rtl_csv, iss_csv, "ibex", iss, report)
+        # construct trace script call from list of strings and call run_cmd()
+        compare_cmd = ("python3 %s/instr_trace_compare.py " % gen_dir) + \
```
Comment (taoliug) [inline dv/uvm/sim.py]: What's the reason for this change? I'd prefer just do a functional call rather than running a bash command.

Comment (udinator) [reply dv/uvm/sim.py]: I changed this to match the rest of the various simulation scripts, and just using a functional call makes argument parsing more wordy, as well as the fact that the argparser  in instr_trace_compare.py will not be needed with that approach

Comment (taoliug) [reply dv/uvm/sim.py]: I think each python script should be able to run independently(argparser is needed in this case), and it can also be included in other python program.  I think we have already avoided running another python script with a bash command. The run_cmd is used for non-python programs.
